evm/heads/heap.go

```diff
@@ -0,0 +1,37 @@
+package heads
+
+import (
+	evmtypes "github.com/smartcontractkit/chainlink-integrations/evm/types"
+)
+
+type headsHeap struct {
+	values []*evmtypes.Head
+}
+
+func (h *headsHeap) Len() int {
+	return len(h.values)
+}
+
+func (h *headsHeap) Swap(i, j int) {
+	h.values[i], h.values[j] = h.values[j], h.values[i]
+}
+
+func (h *headsHeap) Less(i, j int) bool {
+	return h.values[i].Number < h.values[j].Number
+}
+
+func (h *headsHeap) Pop() any {
+	n := len(h.values) - 1
+	old := h.values[n]
```

Comment:
Won't this panic if heap is empty, should we check?

Reply:
I just copied these as-is

Reply:
But panic may be the only option anyways if we are bound to this interface: https://pkg.go.dev/container/heap#Interface